### PR TITLE
Added DbOptions [ECR-839]

### PR DESCRIPTION
**Pull request by vitvakatu**
Abstract way to adjust settings of RocksDB via node config.

TODO:
- [x] Test max_open_files feature
- [x] Add changelog entry (technically, it's breaking change)

### CODE REVIEW

File: exonum/src/storage/rocksdb.rs [hunk 1]
```diff
@@ -42,6 +42,37 @@ pub struct RocksDB {
     db: Arc<rocksdb::DB>,
 }
 
+/// Options for `RocksDB`.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub struct RocksDBOptions {
```
Comment (DarkEld3r) [inline exonum/src/storage/rocksdb.rs]: I propose to name it `DBOptions` (`DbOptions`?).

File: exonum/benches/storage.rs [hunk 2]
```diff
@@ -27,7 +27,7 @@ mod tests {
     use rand::{Rng, thread_rng, XorShiftRng, SeedableRng};
     use tempdir::TempDir;
     use exonum::storage::{Database, MemoryDB};
-    use exonum::storage::{RocksDB, RocksDBOptions};
+    use exonum::storage::{RocksDB, DbOptions};
```
Comment (DarkEld3r) [inline exonum/benches/storage.rs]: Lets merge several `exonum::storage` imports into one.

File: exonum/src/storage/options.rs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+// limitations under the License.
+
+//! Abstract settings for databases.
+
+/// Options for database.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub struct DbOptions {
+    /// Number of open files that can be used by the DB.
+    ///
+    /// Defaults to `-1`, which means opened files are always kept open.
```
Comment (DarkEld3r) [inline exonum/src/storage/options.rs]: I understand that this is an implementation detail of RocksDB, but I don't like such "magic constants". What do you think about using `Options<u32>` here (and converting `None` to the `-1` when needed)?

Comment (vitvakatu) [reply exonum/src/storage/options.rs]: Sounds reasonable. Don't know if we would be able to implement abstract API for any storage though.

**Pull request merged**